lib/database_config.py

```diff
@@ -9,15 +9,33 @@
 
 
 def get_database_config(development_mode=False):
+    # the following options are validated to get database credentials
+    # 1) existence of custom runtime settings Database.... values
+    # 2) VCAP with database credentials
+    # 3) existence of DATABASE_URL
+    #
+    # In case we find MXRUNTIME_Database.... values we don't interfere and
+    # return nothing. VCAP or DATABASE_URL return m2ee configuration
     if any(
         [x.startswith("MXRUNTIME_Database") for x in list(os.environ.keys())]
     ):
+        logging.debug(
+            "Detected database configuration using custom runtime settings."
+        )
         return None
 
     factory = DatabaseConfigurationFactory()
     configuration = factory.get_instance()
 
-    return configuration.get_m2ee_configuration()
+    if configuration:
+        m2ee_config = configuration.get_m2ee_configuration()
+        if m2ee_config is not None and "DatabaseType" in m2ee_config:
```

Comment:
Do you want `if m2ee_config is not None` (i.e. `[]` (an empty list) would return `True` here) or do you actually want `if m2ee_config`? (only "truthy" variables will return `True`)

Reply:
get_m2ee_configuration returns None, "if m2ee_config will work here.